src/Graph.ts

```diff
@@ -0,0 +1,174 @@
+/**
+ * Transform array to graph
+ */
+
+/**
+ * TYPES
+ */
+
+type node = {
+  x: Number;
+  y: Number;
+  value: Number;
+  isExit: Boolean;
+};
+
+type edge = {
+  from: string;
+  to: string;
+  direction: string;
+};
+
+type Graph = {
+  nodes: {
+    [coordinates: string]: node;
+  };
+  edges: Array<edge>;
+};
+
+/**
+ * FUNCTIONS
+ */
+
+export function boardToGraph(board: Array<Array<Number>>): Graph | undefined {
+
+  if (board.length < 1) return undefined;
+
+  let graphFromBoard: Graph = newBlankGraph();
+
+
+  const verticalLines: Number = board.length;
+    const horizontalLines: Number = board[0].length;
+  
+
+    for (let hIndex = 0; hIndex < horizontalLines; hIndex++){
+        for (let vIndex = 0; vIndex < verticalLines; vIndex++){
+
+            graphFromBoard = addBoardBoundaries(graphFromBoard, board, hIndex, vIndex)
+            graphFromBoard = addDirections(graphFromBoard, board, hIndex, vIndex)
+
+        }
+    }
+
+  return graphFromBoard;
+
+}
+
+function newBlankGraph(): Graph {
+  return {
+    nodes: {},
+    edges: [],
+  };
+}
+
+function addBoardBoundaries(graph: Graph, board: Array<Array<Number>>, hIndex: number, vIndex: number): Graph {
+
+    graph.nodes[`${hIndex},${vIndex}`] = {
+        x: hIndex,
+        y: vIndex,
+        value: board[vIndex][hIndex],
+        isExit: isAnExit(board, hIndex, vIndex)
+    }
+
+  return graph;
+}
+
+function addDirections(graph: Graph, board: Array<Array<Number>>, hIndex: number, vIndex: number): Graph {
+
+    graph.edges.push({
+        from: `${hIndex},${vIndex}`,
+        to: `${hIndex-1},${vIndex}`,
+        direction: 'W'
+    })
+
+    graph.edges.push({
+        from: `${hIndex},${vIndex}`,
+        to: `${hIndex+1},${vIndex}`,
+        direction: 'E'
+    })
+
+    graph.edges.push({
+        from: `${hIndex},${vIndex}`,
+        to: `${hIndex},${vIndex-1}`,
+        direction: 'N'
+    })
+
+    graph.edges.push({
+        from: `${hIndex},${vIndex}`,
+        to: `${hIndex},${vIndex+1}`,
+        direction: 'S'
+    })
+
+  return graph;
+}
+
+function isAnExit(board: Array<Array<Number>>, x: Number, y: Number){
+    const firstColumnIndex: Number = 0;
+    const firstRowIndex: Number = 0;
+    const lastColumnIndex: Number = board[0].length - 1;
+    const lastRowIndex: Number = board.length - 1;
+
+    if (x === firstColumnIndex) return true
+    if (y === firstRowIndex) return true
+    if (x === lastColumnIndex) return true
+    if (y === lastRowIndex) return true
+
+    return false
+
+}
+
+/*
+
+TODO: To be removed
+
+obviously all the 0:x and x:0 are exits
+        all board[x, length] and board[length, x] are exits
+        the only admitted directions are N, E, S, W
+
+directions : 
+    x-1 == 'West'
+    x+1 == 'East'
+    y-1 == 'North'
+    y+1 == 'South'
+
+const board = [
```

Comment:
instead of commenting out this code, why don't you convert it to a unit test?

Reply:
It was just a help at the moment, but i understand that it may be a good unit test, i'll include this in tests.
The comment has been removed in the [last commit](https://github.com/marmelab/kuba-integration/commit/52c48e332abaf5397cf6debd1512e2111062ba31)